test/e2e/uninstall_kubelet_test.go

```diff
@@ -0,0 +1,46 @@
+package e2e
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/openshift/windows-machine-config-bootstrapper/pkg/bootstrapper"
+)
+
+func TestKubeletUninstall(t *testing.T) {
+	t.Run("Uninstall Kubelet", testKubeletUninstall)
+}
+
+// testKubeletUninstall tests if WMCB returns an error if the kubelet is uninstalled
+func testKubeletUninstall(t *testing.T) {
+	t.Run("Kubelet service uninstalls", func(t *testing.T) {
+		wmcb, err := bootstrapper.NewWinNodeBootstrapper("", "", "", "", "")
+		require.NoError(t, err, "could not create wmcb")
+
+		err = wmcb.UninstallKubelet()
+		assert.NoError(t, err, "unable to uninstall kubelet service")
+
+		err = wmcb.Disconnect()
+		assert.NoError(t, err, "could not disconnect from windows svc API")
+
+		assert.False(t, svcExists(t, bootstrapper.KubeletServiceName))
+	})
+}
+
+// testUninstallWithoutKubeletSvc tests if WMCB returns an error if the kubelet is uninstalled with no kubelet service present
+func testUninstallWithoutKubeletSvc(t *testing.T) {
+	t.Run("Kubelet service uninstalls", func(t *testing.T) {
+		if svcExists(t, bootstrapper.KubeletServiceName) {
+			t.Skip("Skipping as kubelet service already exists")
+		}
+
+		wmcb, err := bootstrapper.NewWinNodeBootstrapper("", "", "", "", "")
+		require.NoError(t, err, "could not create wmcb")
+
+		err = wmcb.UninstallKubelet()
+		assert.Error(t, err, "no error when attempting to uninstall kubelet without kubelet svc")
+		assert.Contains(t, err.Error(), "kubelet service is not present", "incorrect error thrown")
+	})
+}
```

Comment:
You shouldnt be starting test functions with a `t.Run`. In general `t.Run` should be used to divide separate test cases under a common test parent.
In this code you are introducing essentially 2 tests. `Uninstall kubelet without kubelet service present` and `Uninstall Kubelet`. But these tests are being introduced as `TestBootstrapper/Uninstall_kubelet_without_kubelet_service_present/Kubelet_service_uninstalls` and  `TestKubeletUninstall/Uninstall_Kubelet/Kubelet_service_unistalls`. Theres no need for this complexity.

Instead you can have the two new tests be `TestBootstrapper/Kubelet_service_uninstalls` and `TestKubeletUninstallWithoutKubeletSVC`

Reply:
`You shouldnt be starting test functions with a t.Run.` This statement is true only in this context, where there is only one test occuring within the the function.

Reply:
thanks for pointing this out, modified the test calls.